src/pykoala/plotting/plot_plot.py

```diff
@@ -0,0 +1,418 @@
+           else:
+               #print(xx)
+               #print(y)
+               # print(psym)
+               # print(color) 
+               # print(alpha)
+               # print(label)
+               # print(linewidth)
+               # print(markersize)
+               # print(len(xx),len(y), len(psym), len(color), len(alpha), len(label), len(linewidth), len(markersize))
+               plt.plot(xx[i],y[i], psym[i], color=color[i], alpha=alpha[i], label=label[i], mew=linewidth[i], markersize=markersize[i])
+
+           if ymax == "":
+                    y_max_ = []                
+                    y_max_.extend((y[i][j]) for j in range(len(xx[i])) if (xx[i][j] > xmin and xx[i][j] < xmax) )  
+                    y_max_list.append(np.nanpercentile(y_max_, percentile_max))
+           if ymin == "":
+                    y_min_ = []                
+                    y_min_.extend((y[i][j]) for j in range(len(xx[i])) if (xx[i][j] > xmin and xx[i][j] < xmax) )  
+                    y_min_list.append(np.nanpercentile(y_min_, percentile_min))
+       if ymax is None :
+           ymax = np.nanmax(y_max_list)
+       if ymin is None:
+           ymin = np.nanmin(y_min_list)
+   else:
+       if alpha == "": alpha=1
+       if statistics =="": statistics = True
+       if fcal is None:
+           if np.nanmedian(np.abs(y)) < 1E-10 : 
+               fcal= True 
+               if np.nanmedian(np.abs(y)) < 1E-20 and np.nanvar(np.abs(y)) > 0 : fcal = False
+       if psym == "":
+             plt.plot(x,y, color=color, alpha=alpha,linewidth=linewidth,  linestyle=linestyle)
+       else:
+           plt.plot(x,y, psym, color=color, alpha=alpha, mew=linewidth, markersize=markersize)
+       if ymin is None :
+           y_min_ = []                
+           y_min_.extend((y[j]) for j in range(len(x)) if (x[j] > xmin and x[j] < xmax) )  
+           ymin = np.nanpercentile(y_min_, percentile_min)
+       if ymax is None :
+           y_max_ = []                
+           y_max_.extend((y[j]) for j in range(len(x)) if (x[j] > xmin and x[j] < xmax) )  
+           ymax = np.nanpercentile(y_max_, percentile_max)    
+       
+   plt.xlim(xmin,xmax)                    
+   plt.ylim(ymin,ymax)
+   if ptitle is None: ptitle="Pretty plot"
+   
+   try:   
+       plt.title(ptitle, fontsize=title_fontsize)
+   except Exception:
+       if warnings : print("  WARNING: Something failed when including the title of the plot")
+   
+   plt.minorticks_on()
+   if xlabel is None: xlabel="Wavelength [$\mathrm{\AA}$]"
+   plt.xlabel(xlabel, fontsize=label_axes_fontsize)
+   #plt.xticks(rotation=90)
+   plt.tick_params('both', length=tick_size[0], width=tick_size[1], which='major')
+   plt.tick_params('both', length=tick_size[2], width=tick_size[3], which='minor')
+   plt.tick_params(labelsize=axes_fontsize)
+   plt.axhline(y=ymin,linewidth=axes_thickness, color="k")     # These 4 are for making the axes thicker, it works but it is not ideal...
```

Comment:
Actually, one should define axes rather than calling `plt` directly. I guess
```
ax = plt.gca()
for axis in ['top','bottom','left','right']:
    ax.spines[axis].set_linewidth(axes_thickness)
ax.tick_params(width=axes_thickness)
```
would do the trick.